components/Home/DevIncentives/requests.jsx

```diff
@@ -96,27 +109,44 @@ const getEpochLength = async () => {
   return parseInt(response, 10);
 };
 
-export const canShowCheckpoint = async () => {
+const getEpochDetails = async () => {
   try {
     const epCounter = await getEpochCounter();
-    const epTokenomics = await getEpochTokenomics({
-      lastPoint: Number(epCounter) - 1,
-    });
+    const epTokenomics = await getEpochTokenomics(Number(epCounter) - 1);
     const epochLen = await getEpochLength();
     const blockTimestamp = await getBlockTimestamp();
-    const { endTime } = epTokenomics;
+    const timeDiff = blockTimestamp - epTokenomics.endTime;
 
-    if (blockTimestamp - endTime >= epochLen) {
-      return true;
-    }
-    return false;
+    return { timeDiff, epochLen };
+  } catch (error) {
+    console.error(error);
+  }
+
+  return { timeDiff: 0, epochLen: 0 };
```

Comment:
I'm not understand fully the business logic yet, just want to ensure - in case of an error in `getEpochDetails` we return this `return { timeDiff: 0, epochLen: 0 }`, and `canShowCheckpoint` then will return `true` (`return timeDiff >= epochLen` -> `0 == 0`)

is that expected that in case of an error here we still show `Checkpoint` block?

Reply:
yes as an option, this should work.

or we can do `throw error` in catch block after `console.error(error);`
```
...
} catch (error) {
    console.error(error);
    throw error
}
```